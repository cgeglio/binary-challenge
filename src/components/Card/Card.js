import React from 'react';
import './Card.scss';

export const Card = ({ card }) => {
  return (
    <article>
      <img src={process.env.PUBLIC_URL + `/cardAssets/${card.name_short}.jpg`} alt="Tarot card" />
    </article>
  )
}

export default Card;
