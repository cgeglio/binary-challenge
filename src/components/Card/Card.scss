article {
  margin: 0 25px;
}
