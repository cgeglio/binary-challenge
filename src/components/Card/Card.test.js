import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  it ('should match the snapshot', () => {
    const mockCard = {short_name: 'cu05'};
    const mockNumber = 3;
    const wrapper = shallow(<Card card={mockCard} number={mockNumber} />);
    expect(wrapper).toMatchSnapshot();
  })
})
