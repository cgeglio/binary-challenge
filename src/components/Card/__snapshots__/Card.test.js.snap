// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<article
  className="flip-card fade-in"
>
  <div
    className="flip-card-inner"
  >
    <img
      alt="Tarot card"
      className="flip-card-back"
      src="/cardAssets/undefined.jpg"
    />
    <div
      className="overlay"
    >
      <div
        className="card-details"
      >
        <h3 />
        <h4>
          Meaning:
        </h4>
        <p />
        <h4>
          Description:
        </h4>
        <p />
      </div>
    </div>
    <img
      alt="Tarot card"
      className="flip-card-front"
      src="/cardAssets/GCT-gold.png"
    />
  </div>
</article>
`;
